ports/esp32/machine_timer.c

```diff
@@ -130,20 +136,44 @@ STATIC void machine_timer_enable(machine_timer_obj_t *self) {
 
 STATIC mp_obj_t machine_timer_init_helper(machine_timer_obj_t *self, mp_uint_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
     static const mp_arg_t allowed_args[] = {
-        { MP_QSTR_period,       MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 0xffffffff} },
         { MP_QSTR_mode,         MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 1} },
         { MP_QSTR_callback,     MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+#if MICROPY_PY_MACHINE_HI_RES_TIMER
+        { MP_QSTR_period,       MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+        { MP_QSTR_freq,         MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+#else
+        { MP_QSTR_period,       MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 0xffffffff} },
+#endif
     };
 
     machine_timer_disable(self);
 
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
 
+#if MICROPY_PY_MACHINE_HI_RES_TIMER
+    if (args[3].u_obj != mp_const_none) {
+        mp_float_t freq;
+        if (mp_obj_get_float_maybe(args[3].u_obj, &freq)) {
+	    self->period = (uint64_t) (TIMER_SCALE / freq);
+        } else {
+            mp_raise_TypeError("Frequency must be a number");
+        }
+    } else {
+        mp_float_t period;
+        if (mp_obj_get_float_maybe(args[2].u_obj, &period)) {
```

Comment:
This could alse be `mp_obj_get_float()`

Reply:
Also done. I also made the same change in the same two places in the esp8266 port.